ansible/roles/kube_prometheus_stack/tasks/main.yml

```diff
@@ -0,0 +1,195 @@
+---
+
+- name: Creating namespace
+  kubernetes.core.k8s:
+    name: "{{ kube_prometheus_stack_release_namespace }}"
+    api_version: v1
+    kind: Namespace
+    state: present
+
+# Because of the way Helm handles CRDs, we upgrade them first
+- name: Get kube-prometheus-stack CRDs
+  command: >-
+    helm show crds
+      {{ kube_prometheus_stack_chart_name }}
+      --repo {{ kube_prometheus_stack_chart_repo }}
+      --version {{ kube_prometheus_stack_chart_version }}
+  register: kube_prometheus_stack_crds
+
+- name: Install kube-prometheus-stack CRDs
+  # Use server-side apply because some of the CRDs are too big to fit in the annotation
+  command: kubectl apply --server-side=true --force-conflicts=true -f -
+  args:
+    stdin: "{{ kube_prometheus_stack_crds.stdout }}"
+
+- name: Disable rancher default storage class
+  kubernetes.core.k8s:
+    namespace: "{{ kube_prometheus_stack_release_namespace }}"
+    state: patched
+    definition:
+      kind: StorageClass
+      metadata:
+        name: local-path
+        annotations:
+          storageclass.kubernetes.io/is-default-class: "false"
+
+# variables would need refactoring to let us loop through the data directories nicely
+- name: Create Prometheus hostPath volume in /var/lib/state
+  kubernetes.core.k8s:
+    namespace: "{{ kube_prometheus_stack_release_namespace }}"
+    definition:
+      apiVersion: v1
+      kind: PersistentVolume
+      metadata:
+        name: prometheus-dir
+        labels:
+          app.kubernetes.io/name: prometheus-dir
+      spec:
+        capacity:
+          storage: "{{ prometheus_volume_size }}"
```

Comment:
This needs fudging somehow, it has no effect and there's really no way to chose it